lib/mongo/collection/view/writable.rb

```diff
@@ -93,9 +95,11 @@ def find_one_and_delete(opts = {})
                 txn_num: txn_num,
               ).execute_with_connection(connection, context: context)
             end
-          end.first['value']
+          end.first&.fetch('value', nil)
```

Comment:
An amusing/cute way of spelling this that retains the `[]` operator is `first&.[]('value')`.